src/components/TodoList.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useContext } from 'react';
+import { TodoItem } from './TodoItem';
+import { TodosContext } from '../TodosContext';
+import { Filtering } from '../types/Filtering';
+
+export const TodoList: React.FC = () => {
+  const { todos, filtering } = useContext(TodosContext);
+
+  const visibleTodos = (currentFiltering: Filtering) => {
+    switch (currentFiltering) {
+      case Filtering.ALL:
+        return todos;
+      case Filtering.ACTIVE:
+        return todos.filter((t) => !t.completed);
+      case Filtering.COMPLETED:
+        return todos.filter((t) => t.completed);
+      default:
+        return todos;
+    }
```

Comment:
You can remove `case Filtering.ALL` from this switch, bc in default you have a duplicate, and it will work the same